datafusion/datasource/src/file.rs

```diff
@@ -34,13 +34,51 @@ use datafusion_physical_plan::filter_pushdown::{FilterPushdownPropagation, Pushe
 use datafusion_physical_plan::metrics::ExecutionPlanMetricsSet;
 use datafusion_physical_plan::DisplayFormatType;
 
+use datafusion_physical_expr_common::sort_expr::PhysicalSortExpr;
 use object_store::ObjectStore;
 
 /// Helper function to convert any type implementing FileSource to Arc&lt;dyn FileSource&gt;
 pub fn as_file_source<T: FileSource + 'static>(source: T) -> Arc<dyn FileSource> {
     Arc::new(source)
 }
 
+/// Result of attempting to push down sort ordering to a file source
+#[derive(Debug, Clone)]
+pub enum SortOrderPushdownResult<T> {
+    /// The source can guarantee exact ordering (data is perfectly sorted)
+    Exact { inner: T },
+    /// The source has optimized for the ordering but cannot guarantee perfect sorting
+    /// (e.g., reordered files/row groups based on statistics)
+    Inexact { inner: T },
+    /// The source cannot optimize for this ordering
+    Unsupported,
+}
+
+impl<T> SortOrderPushdownResult<T> {
+    /// Returns true if the result is Exact
+    pub fn is_exact(&self) -> bool {
+        matches!(self, Self::Exact { .. })
+    }
+
+    /// Returns true if the result is Inexact
+    pub fn is_inexact(&self) -> bool {
+        matches!(self, Self::Inexact { .. })
+    }
+
+    /// Returns true if optimization was successful (Exact or Inexact)
+    pub fn is_supported(&self) -> bool {
+        !matches!(self, Self::Unsupported)
+    }
```

Comment:
Are these used anywhere? If not I suggest we keep them out of the public API for now.